### PR TITLE
Feature/ch8810/model deployment design document

### PR DESCRIPTION
**Pull request by popcornylu**

### CODE REVIEW

**Comment by popcornylu** [comment]
Preview
https://docs.primehub.io/docs/next/design/model-deployment

File: docs/design/model-deployment.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+--- | ----- | -----------------------
+`modelDeployment.enabled` | Enable the model deployment | `false`
+
+# Design
+
+## Seldon
+
+[Seldon](https://www.seldon.io/) is a model deployment solution in the community. The reason why we select Seldon as the solution is because it provides a common way to package different framework's by different programming languages into a docker image.
+
+Seldon also provides an operator under `Seldon core` project to manage a `SeldonDeployment` resource and reconcile it to the underlying deployments and services. However, for simplicity, we decide not to use the `SeldonDeployment` resource. Instead, we define `PhDeployment` and the controller generate the underlying `Deployment` and `Service` directly.
```
Comment (ChanYiLin) [inline docs/design/model-deployment.md:57]: Doesn't sound grammatically correct to me...
- Instead, we define `PhDeployment` **along with** the controller **and** generate the underlying `Deployment` and `Service` directly.
- Instead, we define `PhDeployment` and the controller **for generating/to generate** the underlying `Deployment` and `Service` directly.

File: docs/design/model-deployment.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  availableReplicas: 2
+  endpoint: https://primehub.local/deployment/user-defined-postfix/predict
+  history:
+    - time: 2020-03-23T02:03:15Z
+      spec: <PhDeploymentSpec>
+    - time: 2020-03-22T23:45:23Z
+      spec: <PhDeploymentSpec>
+```  
+
+The `PhSchedule` resource has the following children
```
Comment (ChanYiLin) [inline docs/design/model-deployment.md:104]: should be `PhDeployment`

File: docs/design/model-deployment.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+[Seldon](https://www.seldon.io/) is a model deployment solution in the community. The reason why we select Seldon as the solution is because it provides a common way to package different framework's by different programming languages into a docker image.
+
+Seldon also provides an operator under `Seldon core` project to manage a `SeldonDeployment` resource and reconcile it to the underlying deployments and services. However, for simplicity, we decide not to use the `SeldonDeployment` resource. Instead, we define `PhDeployment` and the controller generate the underlying `Deployment` and `Service` directly.
+
+## Custom Resource
+
+A [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) `PhDeployment` is defined for PrimeHub-defined model deployment. The deployment is very similar to the Kubernetes native deployment, the controller would spawn a deployment according to the `PhDeployment`'s spec. The difference is that the spec contains the PrimeHub-specific concept, like User, Group, and InstanceType.
+
+Here is an example of `PhSchedule`.
```
Comment (ChanYiLin) [inline docs/design/model-deployment.md:63]: should be `PhDeployment`

File: docs/design/model-deployment.md (hunk 1 above)
Comment (popcornylu) [reply docs/design/model-deployment.md]: Fixed

File: docs/design/model-deployment.md (hunk 2 above)
Comment (popcornylu) [reply docs/design/model-deployment.md]: Fixed

File: docs/design/model-deployment.md (hunk 3 above)
Comment (popcornylu) [reply docs/design/model-deployment.md]: Fixed

**Pull request merged**